core/core.go

```diff
@@ -59,16 +59,17 @@ func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
 	nodeCtx := common.NodeLocation.Context()
 	domWait := false
 	for i, block := range blocks {
-		isCoincident := c.sl.engine.IsDomCoincident(block.Header())
 
-		// Write the block body to the db.
-		rawdb.WritePendingBlockBody(c.sl.sliceDb, block.Header().Root(), block.Body())
+		isCoincident := c.sl.engine.IsDomCoincident(block.Header())
+		// Write the block body to the lru cache in worker.
+		rawdb.WriteBody(c.sl.sliceDb, block.Hash(), block.Header().NumberU64(), block.Body())
```

Comment:
We can't write to db yet. We don't know if this block is valid. `WritePendingBlockBody()` was the correct method, I believe.

Reply:
yes, changed it to write to the pendingBlockBody.